pkg/services/ngalert/state/cache.go

```diff
@@ -104,13 +106,28 @@ func (c *cache) expandRuleLabelsAndAnnotations(alertRule *ngModels.AlertRule, da
 
 		return expanded
 	}
-
 	return expand(alertRule.Labels), expand(alertRule.Annotations)
 }
 
-func expandTemplate(name, text string, data map[string]string) (result string, resultErr error) {
+// templateCaptureValue represents each value in .Values in the annotations
+// and labels template
+type templateCaptureValue struct {
+	Labels map[string]string
+	Value  *float64
```

Comment:
Mostly for the benefit of my understanding: Are they always float, from previous conversations with @kylebrandt I thought there could be other types too? I guess they are given this is a `NumberValueCapture` as its name implies but would like to clear my understanding.

Reply:
`NumberValueCapture` must contain `float64`. I couldn't see values other than float64 being read in the `eval` package for either `classic.EvalMatch` or `NumberValueCapture`, but I could have missed something!

Reply:
Always *float64 coming from SSE. It will take in other number types from data source queries (e.g. uInts etc) but they are converted to floats on ingestion by SSE.

Reply:
Thanks!